include/tvm/relax/nested_msg.h

```diff
@@ -341,6 +371,57 @@ void DecomposeNestedMsg(Expr expr, NestedMsg<T> msg, FType fvisitleaf) {
   }
 }
 
+/*!
+ * \brief Recursively transform the tuple structure in expr and msgs along with it.
+ *
+ * This function will call ftransleaf for each leaf expression in expr.
+ * This function will throw an error if the nesting structure in msg does not
+ * match the tuple nesting structure in expr.
+ *
+ * \param expr The input expression to be transform.
+ * \param msgs The input messages to guide the transformation.
+ * \param ftransleaf with signature ftransleaf(Expr, Array<NestedMsg<T>>)->Expr
+ * \tparam T the content type of nested msg
+ * \tparam FType The visit function type.
```

Comment:
Would need the docstring for `N` now
```suggestion
 * \param ftransleaf with signature ftransleaf(Expr, Array<NestedMsg<T>>)->Expr
 * \tparam T the content type of nested msg
 * \tparam N ...
 * \tparam FType The visit function type.
```